src/test/ui/suggestions/impl-trait-missing-lifetime-gated.stderr

```diff
@@ -26,17 +26,25 @@ error[E0658]: anonymous lifetimes in `impl Trait` are unstable
   --> $DIR/impl-trait-missing-lifetime-gated.rs:5:31
    |
 LL | fn f(_: impl Iterator<Item = &'_ ()>) {}
-   |                               ^^
+   |                               ^^ expected named lifetime parameter
    |
    = help: add `#![feature(anonymous_lifetime_in_impl_trait)]` to the crate attributes to enable
+help: consider introducing a named lifetime parameter
+   |
+LL | fn f<'a>(_: impl Iterator<Item = &'_'a  ()>) {}
+   |     ++++                            ++
 
 error[E0658]: anonymous lifetimes in `impl Trait` are unstable
   --> $DIR/impl-trait-missing-lifetime-gated.rs:8:31
    |
 LL | fn g(x: impl Iterator<Item = &'_ ()>) -> Option<&'_ ()> { x.next() }
-   |                               ^^
+   |                               ^^ expected named lifetime parameter
    |
    = help: add `#![feature(anonymous_lifetime_in_impl_trait)]` to the crate attributes to enable
+help: consider introducing a named lifetime parameter
+   |
+LL | fn g<'a>(x: impl Iterator<Item = &'_'a  ()>) -> Option<&'_ ()> { x.next() }
+   |     ++++                            ++
```

Comment:
Looking at the resulting code here, the `.shrink_to_hi()` of the span without checking whether the span covers only `&` *or* `'_` causes invalid code to be produced. It also seems like we might also want to replace the anon lifetime in the return type?

Reply:
Replacing the anon lifetime in the return type seems like an outsized change.  I'd rather stabilize this feature sooner, to get rid of the question.

Reply:
Is there anything else I should do?

Reply:
The question is: do you want to take a bit more time in this PR to fix the `'_'a` situation?